core/kernel/tee_ta_manager.c

```diff
@@ -112,15 +112,10 @@ static struct tee_ta_ctx *ts_to_ta_ctx(struct ts_ctx *ctx)
 	panic("bad context");
 }
 
-static bool tee_ta_try_set_busy(struct tee_ta_ctx *ctx)
+static bool tee_ta_try_set_busy_nolock(struct tee_ta_ctx *ctx)
 {
 	bool rc = true;
 
-	if (ctx->flags & TA_FLAG_CONCURRENT)
-		return true;
-
-	mutex_lock(&tee_ta_mutex);
```

Comment:
here is the case I need those _nolock func.
In tee_ta_instance_stats(), it iterates all TA instances and sessions, so mutex_lock(&tee_ta_mutex) before accessing tee_ctxes. During iteration, tee_ta_dump_memstats() will be called to get heap stats and this func will set busy flag since it calls into TA entry func -- dump_mem_stats().
I need tee_ta_try_set_busy_nolock() and tee_ta_clear_busy_nolock() in tee_ta_dump_memstats(), otherwise it will double call mutex_lock(&tee_ta_mutex) in this calling route and cause deadlock.

Reply:
Oh I see, so `tee_ta_mutex` is already locked. That's fine then.